### DISCUSSION TITLE
Get repeated response when using `findOne` using Express

### DISCUSSION THREAD

**Discussion by aroyan** [Automattic/mongoose#11839] [Q&A] [answered]
### This is the code

![code](https://user-images.githubusercontent.com/43630681/170664907-a13061ff-d579-4b18-aa0c-ed4032c7ac0c.jpg)

### This is the database

![mongo](https://user-images.githubusercontent.com/43630681/170664946-c9aa453c-ea1a-43a7-8983-cd0f54de54e1.jpg)

### This is the response when I tried to get request on `first-post` endpoint but I ended up getting `learn-react` data

![postman](https://user-images.githubusercontent.com/43630681/170664960-7ca2f81f-fe9f-4077-9dbd-364884c56d6f.jpg)

### This is the response when I tried to get request on `learn-node` endpoint but I ended up getting `learn-react` data

![server](https://user-images.githubusercontent.com/43630681/170664975-2a2c55c1-a686-4bee-8167-15f2a51df758.jpg)

I wonder if I got the wrong syntax or something? I've followed the documentation but it still giving me the same result. Is there any way to solve this? Thank you

**Comment by Uzlopak** [comment, marked as answer]
probably articleschema does not contain name and then mongoose filters on findOne the name field and then it is just returning the first document in the collection.

**Reply by aroyan** [reply]
Thank you, it's working now. I've been looking for a day to figure out what's wrong it turned out my schema.
